modules/Bio/EnsEMBL/Variation/DBSQL/VariationFeatureAdaptor.pm

```diff
@@ -2348,4 +2348,196 @@ sub fetch_all_by_location_identifier {
   return $vfs;
 }
 
+=head2 fetch_by_spdi_notation
+
+    Arg[1]      : String $spdi
+    Example     : my $spdi = 'NC_000016.9:68644751:T:A';
+                  $vf = $vf_adaptor->fetch_by_spdi_notation($spdi);
+    Description : Parses a SPDI notation and tries to create a VariationFeature object
+    based on the notation. The object will have a Variation and Alleles attached.
+    ReturnType  : Bio::EnsEMBL::Variation::VariationFeature, undef on failure
+    Exceptions  : thrown on error
+    Caller      : general
+    Status      : Stable
+
+=cut
+
+sub fetch_by_spdi_notation{ 
+  my $self = shift; 
+
+  my ($spdi, $user_slice_adaptor, $user_transcript_adaptor, $multiple_ok, $replace_ref) = rearrange([qw(
+    SPDI
+    SLICE_ADAPTOR
+    TRANSCRIPT_ADAPTOR
+    MULTIPLE_OK
+    REPLACE_REF
+  )], @_);
+
+  ########################### Check & split input ########################### 
+  my ($sequence_id, $position, $deleted_seq, $inserted_seq) = split /:/, $spdi; 
+
+  # count number of elements   
+  my $count_separator = () = $spdi =~ m/\:/g; 
+
+  throw ("Could not parse the SPDI notation $spdi")
+    unless ($count_separator == 3);  
+
+  # strip version number from reference 
+  if($sequence_id =~ m/\./i){
+    $sequence_id =~ s/\.\d+//g;
+  } 
+
+  # check if position is a number 
+  my $check_position = $position =~ m/^\d+\z/i; 
+
+  # check deleted sequence (number, string)   
+  my $check_deleted_seq = $deleted_seq =~ m/^\w+$/;
+  # check deleted sequence (string)  
+  my $check_deleted_seq_letters = $deleted_seq =~ m/^([A-Z]+)$/i;  
+
+  # check inserted sequence (number, string) 
+  my $check_inserted_seq = $inserted_seq =~ m/^\w+$/;
+  # check inserted sequence (string) 
+  my $check_inserted_seq_letters = $inserted_seq =~ m/^([A-Z]+)$/i;  
+
+  throw ("Could not parse the SPDI notation $spdi")  
+    unless (defined($sequence_id) && $sequence_id ne '' && defined($position) && $check_position ne '' && $position ne '' && defined($deleted_seq) && defined($inserted_seq) 
+    && ($check_deleted_seq || $deleted_seq eq '') && ($check_inserted_seq || $inserted_seq eq '')); 
+
+  #######################  extract genomic coordinates and reference seq allele  ####################### 
+
+  my ($start, $end, $strand, $ref_allele, $alt_allele); 
+
+  $strand = 1; ## strand should be genome strand for SPDI genomic notation
+
+  # Get a slice adaptor to enable check of supplied reference allele
+  my $slice_adaptor = $user_slice_adaptor || $self->db()->dnadb()->get_SliceAdaptor(); 
+  my $slice = $slice_adaptor->fetch_by_region('chromosome', $sequence_id ) || $slice_adaptor->fetch_by_region(undef, $sequence_id);  
```

Comment:
It would be better to check a slice is returned and throw if not. Otherwise, a error is returned:

Can't call method "seq" on an undefined value at ensembl-variation/modules/Bio/EnsEMBL/Variation/DBSQL/VariationFeatureAdaptor.pm line 2497